File: module_music/src/main/java/com/yyxnb/module_music/viewmodel/MusicViewModel.java
package com.yyxnb.module_music.viewmodel;

import android.arch.lifecycle.LiveData;

import com.yyxnb.module_music.bean.MusicBean;
import com.yyxnb.module_music.bean.MusicLocalBean;
import com.yyxnb.module_music.bean.MusicRecordBean;
import com.yyxnb.module_music.config.DataConfig;
import com.yyxnb.module_music.db.MusicDao;
import com.yyxnb.module_music.db.MusicDatabase;
import com.yyxnb.module_music.db.MusicLocalDao;
import com.yyxnb.module_music.db.RecordDao;
import com.yyxnb.network.BaseViewModel;

import java.util.List;

public class MusicViewModel extends BaseViewModel {

    private MusicDao musicDao = MusicDatabase.getInstance().musicDao();
    private MusicLocalDao localDao = MusicDatabase.getInstance().musicLocalDao();
    private RecordDao recordDao = MusicDatabase.getInstance().recordDao();

    // 列表
    public LiveData<List<MusicBean>> getMusicData() {
        return musicDao.liveMusicList();
    }

    //本地
    public LiveData<List<MusicLocalBean>> getLocalMusicData() {
        return localDao.liveMusicList();
    }

    //最近一条历史记录/播放着的
    public LiveData<MusicRecordBean> getRecordData() {
        return recordDao.liveLastRecord();
    }

    //历史记录
    public LiveData<List<MusicRecordBean>> getRecordListData() {
        return recordDao.liveRecordAll();
    }

    public void reqMusicData() {

        if (null == musicDao.liveMusicList()) {
        }
        musicDao.deleteAll();
        musicDao.insertItems(DataConfig.getMusicBeans());
//        mMusicData = musicDao.liveMusicList();
    }


}

File: module_music/src/main/java/com/yyxnb/module_music/viewmodel/MusicViewModel.kt
package com.yyxnb.module_music.viewmodel

import androidx.lifecycle.LiveData
import com.yyxnb.module_music.bean.MusicBean
import com.yyxnb.module_music.bean.MusicLocalBean
import com.yyxnb.module_music.bean.MusicRecordBean
import com.yyxnb.module_music.config.DataConfig.musicBeans
import com.yyxnb.module_music.db.MusicDatabase.Companion.instance
import com.yyxnb.network.BaseViewModel

class MusicViewModel : BaseViewModel() {

    private val musicDao = instance!!.musicDao()
    private val localDao = instance!!.musicLocalDao()
    private val recordDao = instance!!.recordDao()

    // 列表
    val musicData: LiveData<List<MusicBean>>
        get() = musicDao.liveMusicList()

    //本地
    val localMusicData: LiveData<List<MusicLocalBean>>
        get() = localDao.liveMusicList()

    //最近一条历史记录/播放着的
    val recordData: LiveData<MusicRecordBean>
        get() = recordDao.liveLastRecord()

    //历史记录
    val recordListData: LiveData<List<MusicRecordBean>>
        get() = recordDao.liveRecordAll()

    fun reqMusicData() {
        if (null == musicDao.liveMusicList()) {
        }
        musicDao.deleteAll()
        musicDao.insertItems(musicBeans!!)
        //        mMusicData = musicDao.liveMusicList();
    }
}